### ISSUE TITLE
WARNING: method definition ... declares type variable AD but does not use it

### ISSUE THREAD

**Issue by dpo** [JuliaSmoothOptimizers/ADNLPModels.jl#159]
With Julia 1.9 and ADNLPModels 0.6.2, I'm getting
```julia
┌ ADNLPModels [54578032-b7ea-4c30-94aa-7cbd1cce6c9a]
│  WARNING: method definition for #ADNLSModel!#251 at /home/runner/.julia/packages/ADNLPModels/gdqYX/src/nls.jl:149 declares type variable AD but does not use it.
│  WARNING: method definition for #ADNLSModel#287 at /home/runner/.julia/packages/ADNLPModels/gdqYX/src/nls.jl:570 declares type variable AD but does not use it.
│  WARNING: method definition for #ADNLSModel!#290 at /home/runner/.julia/packages/ADNLPModels/gdqYX/src/nls.jl:600 declares type variable AD but does not use it.
└ 
```

https://github.com/JuliaSmoothOptimizers/RegularizedProblems.jl/actions/runs/5180679050/jobs/9335162380?pr=41#step:6:680

**Comment by tmigot** [comment]
I think this is thé same as #156 and fixed in #157

**Comment by dpo** [comment]
Great, thanks. Maybe we just need a bugfix release then.
